### ISSUE TITLE
must I use the `train.py` before using `train_cluster_classifier.py` ?

### ISSUE THREAD

**Issue by adaxigua** [wpeebles/gangealing#9]
**Question:**
Is the  `train_cluster_classifier.py`  independent of the `train.py` ?

**Description**
When I want to train a cluster model, I find the following code In `train_cluster_classifier.py`

~~~
# Load pre-trained STN, Generator and LL (required):
    print(f"Loading model from {args.ckpt}")
    ckpt = torch.load(args.ckpt, map_location=lambda storage, loc: storage)
    generator.load_state_dict(ckpt["g_ema"])
    t_ema.load_state_dict(ckpt["t_ema"])
    ll.load_state_dict(ckpt["ll"])
~~~

steps:
1. train a common model by the `train.py` to get a weight.pt
2. using the  weight.pt, enhance the model by the `train_cluster_classifier.py` 

Are the steps right?
In other words, must I use the `train.py` before using `train_cluster_classifier.py` ?

thanks！

**Comment by wpeebles** [comment]
Yep, that's correct. `train.py` will train the Spatial Transformer Network (with multiple output heads for the clustering case). Then, passing that checkpoint into `train_cluster_classifier.py` via `--ckpt` will train a ResNet classifier that figures out which of the STN's output heads should be used when given a new input image (STN weights are frozen in `train_cluster_classifier`).
